src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/XSharedTransformer.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.AstNodes;
+using XamlX;
+using XamlX.Ast;
+using XamlX.Emit;
+using XamlX.IL;
+using XamlX.Transform;
+using XamlX.TypeSystem;
+
+namespace Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.Transformers;
+
+class XSharedTransformer : IXamlAstTransformer
+{
+    public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode node)
+    {
+        var types = context.GetAvaloniaTypes();
+        switch (node)
+        {
+            case XamlPropertyAssignmentNode { Values.Count: 2 } pa when
+                    pa.Property.Name == "Content"
+                    && pa.Property.DeclaringType == types.ResourceDictionary
+                    && pa.Values[1] is XamlAstConstructableObjectNode co
+                    && TryGetSharedDirective(co, out var directive)
+                    :
```

Comment:
Nit: formatting, I don't think we ever have have a single `:` on a line by itself, place it on the line above.